### DISCUSSION TITLE
What magic is involved to load 566 packages in 0.5 seconds ?

### DISCUSSION THREAD

**Discussion by ghost** [jcs090218/jcs-emacs#44] [General]
Once more awesome config !

1) I had to copy the ~/.emacs.d/fonts to my ~/.loacal/share/fonts/ and update font cache with fc-cache-v, other wise on every startup it said done installing fonts

2) I wanted to open a file, I tried C-x C-f, but Ctrl+Alt+l took noticeable delay.

3) Is it correct to assume this is a config for expert users ? is it possible somehow to disable the keys and revert back to pure vanilla emacs bindings ?

**Comment by jcs090218** [comment]
> 1. I had to copy the ~/.emacs.d/fonts to my ~/.loacal/share/fonts/ and update font cache with fc-cache-v, other wise on every startup it said done installing fonts

This is unexpected. For the fonts, I am using [use-ttf](https://github.com/jcs-elpa/use-ttf) to install fonts across all OSs. Did you change the font name? See variable `use-ttf-install-fonts`. 🤔 

> 2. I wanted to open a file, I tried C-x C-f, but Ctrl+Alt+l took noticeable delay.

Use <kbd>Alt</kbd>+<kbd>f</kbd> instead.

`Ctrl+Alt+l` delay is from the upstream `treemacs`; there isn't really much I can do. 😓

> 3. Is it correct to assume this is a config for expert users ?

I think the answer is **yes**.

Before using this configuration, you need to know the answer for these questions. 🤔 

- Why do you use Emacs? Is it worth it?
- Why is Emacs created? (Knowing the philosophy)
- Know the **pros and cons** between heavily configured config ([doomemacs](https://github.com/doomemacs/doomemacs) and [spacemacs](https://www.spacemacs.org/)) vs Vanilla Emacs config ([centaur-emacs](https://github.com/seagle0128/.emacs.d) and [purcell's config](https://github.com/purcell/emacs.d)).
- What are Emacs' flaws? How can we improve it?

These will help you understand why `jcs-emacs` is designed this way and what you can get from here. Your question from the title (What magic is involved in loading 566 packages in 0.5 seconds ?) will be unraveled by one of the questions as well. ;)

> is it possible somehow to disable the keys and revert back to pure vanilla emacs bindings ?

**Unlike Vim's philosophy**, Emacs tends to be **customized**; therefore, I don't recommend people keep using pure vanilla emacs bindings. 😅 I know many people are learning vanilla emacs bindings, but that doesn't fit Emacs philosophy. Emacs' default key bindings are awful, you can google `Emacs pinky`, and you might know why. 🤔 

I suggest overwriting the key you don't like; then your life will be much happier! :D
